File: ownership/ownership.py
#! /usr/bin/env python3

import threading
import time

def background_work(v):
    time.sleep(0.1)
    v.append(42)

def main():
    v = []
    assert len(v) == 0
    thread = threading.Thread(target=background_work, args=[v])
    thread.start()
    # OOPS: swap these two lines
    thread.join()
    assert 42 in v

main()

File: ownership/ownership.rs
use std::thread;
use std::thread::sleep;
use std::time::Duration;

fn background_work(mut v: Vec<i32>) -> Vec<i32> {
    sleep(Duration::from_millis(100));
    v.push(42);
    v
}

fn main() {
    let mut v = Vec::new();
    assert_eq!(v.len(), 0);
    let thread = thread::spawn(|| background_work(v));
    // OOPS: swap these two lines
    v = thread.join().unwrap();
    assert_eq!(v.len(), 1);
}
